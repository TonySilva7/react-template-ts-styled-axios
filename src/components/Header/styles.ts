import styled from 'styled-components';

export const Wrapper = styled.header`
	display: flex;
	align-items: center;
	justify-content: space-between;

	background: ${({ theme }) => theme.colors.primary};

	a {
		display: flex;
		align-items: center;
	}
`;
