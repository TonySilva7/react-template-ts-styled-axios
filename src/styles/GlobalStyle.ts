import { createGlobalStyle } from 'styled-components';

// Global styles
export const GlobalStyle = createGlobalStyle` 
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  
  html {
    font-size: 62.5%;

    body, #root {
      height: 100%;
    }
    
    body {
      font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
      'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
      sans-serif;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
    }

    a {
      text-decoration: none;
    }

    li {
      list-style: none;
    }

    button {
      cursor: pointer;
      border: none;
    }

    code {
      font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
        monospace;
    }
  }
`;
