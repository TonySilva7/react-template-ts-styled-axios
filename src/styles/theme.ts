export const theme = {
	colors: {
		primary: '#00bcd4',
		secondary: '#ff9800',
		accent: '#ff5722',
		error: '#f44336',
		warning: '#ffeb3b',
		info: '#2196f3',
		success: '#4caf50',
		background: '#fafafa',
		surface: '#ffffff',
		text: '#212121',
		// onPrimary: '#000000',
		// onSecondary: '#000000',
		// onAccent: '#000000',
		// onError: '#ffffff',
		// onWarning: '#000000',
		// onInfo: '#000000',
		// onSuccess: '#ffffff',
		// onBackground: '#000000',
		// onSurface: '#000000',
		// onText: '#ffffff',
	},
	spacing: {
		none: 0,
		extraTight: 4,
		tight: 8,
		snug: 12,
		regular: 16,
		loose: 20,
		extraLoose: 24,
	},
	borderRadii: {
		none: 0,
		extraSmall: 2,
		small: 4,
		regular: 8,
		large: 16,
		extraLarge: 32,
	},
	breakpoints: {
		mobile: 0,
		phablet: 550,
		tablet: 750,
		desktop: 1000,
		hd: 1200,
	},
	font: {
		sizes: {
			xsmall: 12,
			small: 14,
			medium: 16,
			large: 20,
			xlarge: 24,
		},
		weights: {
			light: 300,
			regular: 400,
			medium: 500,
			bold: 700,
		},
	},
};
