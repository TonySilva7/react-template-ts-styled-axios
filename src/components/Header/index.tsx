import { Wrapper } from './styles';

export default function Header() {
	return (
		<Wrapper>
			<h1>Header</h1>
		</Wrapper>
	);
}
