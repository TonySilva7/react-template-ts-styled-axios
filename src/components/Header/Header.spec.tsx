import { render, screen } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import Header from '.';
import { theme } from '../../styles/theme';

export const renderTheme = (children: ReactNode) => {
	return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

test('Header renders correctly', () => {
	const { debug } = renderTheme(<Header />);
	// debug();
	const headingHome = screen.getByRole('heading', { name: 'Header' }).parentElement;
	expect(headingHome).toBeInTheDocument();
	expect(headingHome).toHaveStyle({
		backgroundColor: theme.colors.primary,
	});

	// const { container } = renderTheme(<Header />);
	// expect(container).toMatchSnapshot();
});
