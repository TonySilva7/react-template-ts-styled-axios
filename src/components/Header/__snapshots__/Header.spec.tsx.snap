// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="sc-bdfBQB fRDkPb"
  >
    <h1>
      Header
    </h1>
  </header>
</div>
`;
